src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombiner.java

```diff
@@ -136,4 +257,23 @@ private TotalHits getTotalHits(final List<TopDocs> topDocsPerSubQuery, final lon
         }
         return new TotalHits(maxHits, totalHits);
     }
+
+    // Tie-breaker to merge multiple top docs
+    private Comparator<ScoreDoc> getTieBreaker() {
+        final Comparator<ScoreDoc> SORTING_TIE_BREAKER = (o1, o2) -> {
+            int scoreComparison = Double.compare(o1.score, o2.score);
+            if (scoreComparison != 0) {
+                return scoreComparison;
+            }
+
+            int docIdComparison = Integer.compare(o1.doc, o2.doc);
+            if (docIdComparison != 0) {
+                return docIdComparison;
+            }
+
+            // When duplicate result found then both score and doc ID are equal, return 1
+            return (o1.score == o2.score && o1.doc == o2.doc) ? 1 : 0;
```

Comment:
I think in core that use shard id as a third params to run the sort in case doc id and score are same. Do we have shard id info here to do the same?

Reply:
Actually No, at the normalization phase we do not have shard info. It comes in the fetch phase in multiple shard scenario.

To help better understand the situation let me put a example here.
Consider a result which is a part of subquery 1 and subquery 2 provided we have 2 subqueries under the hybrid clause.
The result before TopDocs.merge will look like
<docId, Magic number, -1, [1]>
<docId, Magic number, -1, [1]>
<docId, Magic number, -1, [1]>

Reply:
got it, then we should be good with what you do have in PR now